### PR TITLE
Compliance update

### PR DESCRIPTION
**Pull request by usmnzain**
Updated serv to support latest version of riscv-arch-test (v2.6.1)

### CODE REVIEW

**Comment by olofk** [comment]
Hi @usmnzain Finally got some time to look at this. Thanks for your work. It seems to work fine for running the `I` tests but I think we need some more to run e.g. the `privilege` and `M` tests.

I looked at how other targets handles this and it looks like they all do it differently, but I think the simplest thing is to do like e.g. `grift` and just create the `privilege`, `M` and `Zifencei` dirs alongside the `I` dir with slightly different Makefile.include inside. Can you do that?

Another thing I want to change in your patch that it looks like you are building the simulation model from inside the serv repo (which is why I believe you also added the .gitignore). The recommended way is to build it in a separate workspace directory. I would like you to drop the .gitignore file and use `TARGET_SIM=$WORKSPACE....` in the compliance test instructions.

**Comment by usmnzain** [comment]
Hi @olofk Totally forgot to add other tests. I finally have some time on my hand to look into this. I will add the tests and review other changes you mentioned and update the PR.

**Comment by olofk** [comment]
Sounds good. Happy to take another look after that

**Comment by usmnzain** [comment]
@olofk The PR has been updated to support compliance tests for `M`, `Zifencei` and `privilege`.

**Comment by usmnzain** [comment]
Ran the compliance for `Zifencei` and there are few tests that are failing.

```Check ebreak                    ... FAIL
Check ecall                     ... FAIL
Check misalign1-jalr-01         ... OK
Check misalign2-jalr-01         ... FAIL
Check misalign-beq-01           ... FAIL
Check misalign-bge-01           ... FAIL
Check misalign-bgeu-01          ... FAIL
Check misalign-blt-01           ... FAIL
Check misalign-bltu-01          ... FAIL
Check misalign-bne-01           ... FAIL
Check misalign-jal-01           ... FAIL
Check misalign-lh-01            ... OK
Check misalign-lhu-01           ... OK
Check misalign-lw-01            ... OK
Check misalign-sh-01            ... OK
Check misalign-sw-01            ... OK 
```

File: README.md [hunk 1]
```diff
@@ -76,23 +76,25 @@ Other applications can be tested by compiling and converting to bin and then hex
...
 
-    git clone https://github.com/riscv/riscv-compliance --branch 1.0
+    git clone https://github.com/riscv-non-isa/riscv-arch-test.git
 
 To run the RISC-V compliance tests, we need to supply the SERV-specific support files and point the test suite to where it can find a target to run (i.e. the previously built Verilator model)
 
 Run the compliance tests
 
-    cd riscv-compliance && make TARGETDIR=$SERV/riscv-target RISCV_TARGET=serv RISCV_DEVICE=rv32i RISCV_ISA=rv32i TARGET_SIM=$WORKSPACE/build/servant_1.1.0/verilator_tb-verilator/Vservant_sim
+    cd riscv-arch-test && make TARGETDIR=$SERV/riscv-target RISCV_TARGET=serv RISCV_DEVICE=I TARGET_SIM=$SERV/build/servant_1.1.0/verilator_tb-verilator/Vservant_sim
```
[the remark this answers is not in the document]
Comment (usmnzain) [reply README.md]: fixed in latest commit

**Comment by olofk** [comment]
Perfect! Many thanks for your contributions. Squashed and merged!

**Pull request merged**